### DISCUSSION TITLE
Creating score without MusicXML input, via an API.

### DISCUSSION THREAD

**Discussion by ailac22** [opensheetmusicdisplay/opensheetmusicdisplay#1047] [Q&A] [answered]
Hello,

I've been searching how to do this with this library but so far I've seen no easy way or info about how to create a score programmatically. For instance, VexFlow has the EasyScore API. How is this supposed to work in this library? I have not been able to even create my own `load` function on the library by inputing just a custom MusicSheet created by me, due to problems with the EngravingRules.

Thank you

**Comment by sschmidTU** [comment, marked as answer]
Hi! First of all, OSMD is explicitly a library for displaying MusicXML files, not for creating sheet music.
Most music creating programs export to MusicXML. Which one(s) are you using?

If you still want to use OSMD to create scores, you can do that by creating a MusicSheet (osmd.sheet), then calling `osmd.render()`, see this comment:
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/752#issuecomment-634738966

This comment will show you an example from our project OSME, which also created sheets in OSMD directly. But it will take a while to get used to this way of creating music, probably much longer than another language or music program which can export to MusicXML.

What were your problems with EngravingRules? All you have to do is create a MusicSheet, as i said.
